lib/flake.nix

```diff
@@ -0,0 +1,97 @@
+{
+  description = "DevOS environment configuriguration library.";
+
+  inputs =
+    {
+      deploy.url = "github:serokell/deploy-rs";
+      deploy.inputs = {
+        utils.follows = "flake-utils";
+      };
+      devshell.url = "github:numtide/devshell";
+      flake-utils.url = "github:numtide/flake-utils/flatten-tree-system"; # TODO: pull in check api for simple flake
+      
+    }
+
+  outputs = { self, nixpkgs, deploy, devshell, flake-utils, ... }: let
+
+    inherit (nixpkgs) lib;
+
+    jobs = {
+      test = import ./tests {
+        inherit (self) lib;
+        inherit (nixpkgs) pkgs;
+      };
+      doc = import ./docs { 
+        inherit (self) lib;
+        inherit (nixpkgs) pkgs;
+      };
+      updateTemplateRepos = import ./jobs/updateTemplateRepos.nix {
+        inherit (self) rev;
+        inherit (nixpkgs) pkgs;
+        repos = [
+          "github.com/divnix/devenv"
+          "github.com/divnix/bizenv"
+        ];
+      };
+    };
+
+
+  in {
+
+    lib = lib.extend (final: prev:
+
+      {
+        attrs = import ./attrs.nix { lib = prev; } // prev.attrs;
+        lists = import ./lists.nix { lib = prev; } // prev.lists;
+        strings = import ./strings.nix { lib = prev; } // prev.strings;
+
+        nixosSystemPlus = import ./nixos-system-plus { lib = final; };
```

Comment:
I realized: this is actually not `devos`. It's more like `lib.NixosSystem` on steroids.
Some aspects of what we put here might go into upstream or into [nixos-genertors](https://github.com/nix-community/nixos-generators) (or come from there).